app/models/solid_queue/claimed_execution.rb

```diff
@@ -63,7 +63,7 @@ def perform
     if result.success?
       finished
     else
-      failed_with(result.error)
+      failed_with(result.error, reraise: true)
```

Comment:
Looks like this is the only case where we'd want to reraise (the other failures are "artificial"), and we do that outside the transaction to ensure the failure is registered in the system. If that's true, we could avoid the `reraise` parameter and reraise here instead: 
```suggestion
      failed_with(result.error)
      raise error
```
And then we don't need to modify the rest?

Reply:
Good call. Should be updated now